packages/angular/src/tracing.ts

```diff
@@ -101,6 +111,38 @@ export class TraceService implements OnDestroy {
     }),
   );
 
+  // The ResolveEnd event is fired when the Angular router has resolved the URL
+  // and activated the route. It holds the new resolved router state and the
+  // new URL.
+  // Only After this event, the route is activated, meaning that the transaction
+  // can be updated with the parameterized route name before e.g. the route's root
+  // component is initialized. This should be early enough before outgoing requests
+  // are made from the new route.
+  // In any case, this is the earliest stage where we can reliably get a resolved
+  //
+  public resEnd: Observable<Event> = this._router.events.pipe(
+    filter(event => event instanceof ResolveEnd),
+    tap(event => {
+      const ev = event as ResolveEnd;
+
+      const params = getParamsOfRoute(ev.state.root);
+
+      // ev.urlAfterRedirects is the one we prefer because it should hold the most recent
+      // one that holds information about a redirect to another route if this was specified
+      // in the Angular router config. In case this doesn't exist (for whatever reason),
+      // we fall back to ev.url which holds the primarily resolved URL before a potential
+      // redirect.
+      const url = ev.urlAfterRedirects || ev.url;
+      const route = getParameterizedRouteFromUrlAndParams(url, params);
+
+      const transaction = this._activeTransaction;
+      if (transaction && transaction.metadata.source === 'url') {
```

Comment:
My reasoning for checking for the transaction source here: If users started a custom transaction and a navigation occurs, we shouldn't overwrite the transactions' name. Does this make sense?

Reply:
Sure we can revisit this for the next major

Reply:
Do we have this situation in other routing instrumentations? If yes, we should make a consistent decision across all of them

Reply:
Added a TODO to revisit this in v8
[23242aa](https://github.com/getsentry/sentry-javascript/pull/5416/commits/23242aa88d02054c0f128c04389e0da476cc77ea)